pubsub/README.rst

```diff
@@ -13,54 +13,164 @@ Installation
-- an optionally infinite async subscription poller has been introduced
+Currently we have only implemented an asyncio version of SubscriberClient as
+the subscription pattern seems to have a greater benefit from the introduction
+of async programming.
 
-The upstream pubsub implementation has been pinned to 0.25 since newer versions
-can cause odd errors when combined with ``asyncio`` processing. At the time of
-this writing, a new version of the upstream pubsub implementation is being
-generated -- once we've tested that version, will update this library.
 
 Here's the rough usage pattern for subscribing:
 
 .. code-block:: python
 
-    import gcloud.aio.pubsub as pubsub
+    from gcloud.aio.pubsub import SubscriberClient
+    from google.cloud.pubsub_v1.subscriber.message import Message
 
-    subscriber = pubsub.Client(project)
-    topic = subscriber.topic(topic)
-    subscription = topic.subscription(subscription)
+    client = SubscriberClient()
+    # create subscription if it doesn't already exist
+    client.create_subscription('subscription_name', 'topic_name')
 
-    async for job_id, message in subscription.poll():
-        # do something with message
-        await subscription.acknowledge([job_id])
+    async def message_callback(message: Message) -> None:
+        try:
+            # Just an example: process the message however you need to here...
+            result = handle(message)
+            await upload_result(result)
+        except Exception:
+            message.nack()
+        else:
+            message.ack()
 
-A couple additional features you might want to play with:
+    # subscribe to the subscription, receiving a Future that acts as a keepalive
+    keep_alive = client.subscribe('subscription_name', message_callback)
+
+    # have the client run forever, pulling messages from this subscription, passing
+    # them to the specified callback function, and wrapping it in an asyncio task.
+    client.run_forever(keep_alive)
+
+
+Configuration
+-------------
+
+Our create_subscription method is a thing wrapper and thus supports all keyword
+configuration arguments from the official pubsub client which you can find in the
+`Official Google Documentation <https://github.com/googleapis/google-cloud-python/blob/11c72ade8b282ae1917fba19e7f4e0fe7176d12b/pubsub/google/cloud/pubsub_v1/gapic/subscriber_client.py#L236>`_.
+
+When subscribing to a subscription you can optionally pass in a FlowControl and/or Scheduler instance.
+
+.. code-block:: python
+
+    example_flow_control = FlowControl(
+        max_messages=1,
+        resume_threshold=0.8,
+        max_request_batch_size=1,
+        max_request_batch_latency=0.1,
+        max_lease_duration=10,
+    )
+
+    keep_alive = client.subscribe(
+        'subscription_name',
+        message_callback,
+        flow_control=example_flow_control
+    )
+
+Understand how modifying FlowControl affects how your pubsub runtime will operate can
```

Comment:
Grammar: `Understanding`, `can *be* confusing,`.

Reply:
words, hard 
🦆